lib/build.ts

```diff
@@ -187,3 +194,173 @@ export async function runBuildTask(
+	task: BuildTask,
+	docker: Dockerode,
+): Promise<boolean> {
+	let imageReferences: string[];
+
+	const checkHasV1Manifest = async (repository: string) => {
+		try {
+			const manifest = await getManifest(docker.modem, repository);
+			return manifest.Descriptor.mediaType === MEDIATYPE_MANIFEST_V1;
+		} catch {
+			// eat the exception... yummy!
+		}
+
+		console.log(
+			`Image manifest data unavailable for ${repository}. Platform support will be assumed.`,
+		);
+		// at this time, treat any errors as false (no v1 manifest)
+		// This will cause the `--platform` flag to get passed as if the image has
+		// platform support.
+		return false;
+	};
+
+	if (task.imageName) {
+		imageReferences = [task.imageName + !!task.tag ? `:${task.tag}` : ''];
+	} else {
+		if (!task.dockerfile) {
+			// Sanity check
+			console.log('Build task does not have an associated Dockerfile');
+			return false;
+		}
+		const { DockerfileParser } = await import('dockerfile-ast');
+		const parsedDockerfile = DockerfileParser.parse(task.dockerfile);
+		const dockerInstructions = parsedDockerfile.getInstructions();
+		const fromInstructions = dockerInstructions.filter(
+			(inst) => inst.getKeyword() === 'FROM',
+		);
+		if (fromInstructions.length === 0) {
+			// Sanity check
+			console.log(
+				'Build task associated Dockerfile does no reference any images',
+			);
+			return false;
+		}
+		imageReferences = [];
+		fromInstructions.forEach((inst) => {
+			const fromLineArguments = inst.getArguments();
+			if (fromLineArguments.length === 0) {
+				// bail out, no image specified?
+				return;
+			}
+
+			let imgRef = fromLineArguments[0].getValue();
+
+			// hacky way to ignore the --platform flag in the FROM line
+			// In that case, just get the next argument :-p
+			if (imgRef.startsWith('--platform')) {
+				if (fromLineArguments.length < 2) {
+					// bail out, no image specified?
+					return;
+				}
+				imgRef = fromLineArguments[1].getValue();
+			}
+			imageReferences.push();
+		});
+	}
+
+	const v1Images: string[] = [];
+	const v2Images: string[] = [];
+
+	await Promise.all(
+		imageReferences.map(async (r) => {
+			const hasV1Manifest = await checkHasV1Manifest(r);
+			if (hasV1Manifest) {
+				v1Images.push(r);
+			} else {
+				v2Images.push(r);
+			}
+		}),
+	);
+
+	if (v1Images.length === 0) {
+		// Only v2 images, let Docker receive `--platform`
+		return true;
+	} else if (v2Images.length > 0) {
+		// Mixed v1 and v2 images
+		console.log(
+			'Warning: Dockerfile contains a mixes images that require an architecture selection with ones that do not support it.\n' +
+				'The following images do not support architecture selection:\n-' +
+				v1Images.join('\n-') +
+				'\n\n' +
+				'The following images require architecture selection:\n-' +
+				v2Images.join('\n-') +
+				'\n\n' +
+				'To avoid a possible build error, the CLI has disabled platform selection. ' +
+				'As a result, the architecture of the machine where the Docker Engine ' +
```

Comment:
IMO this is the only place where it is appropriate to use `architecture` as it is a classification of the workstation, everywhere else we should use `platform` to align with Docker vernacular.

Reply:
Ok, I think I have all the cases sorted out now.